src/parse.y

```diff
@@ -206,8 +206,14 @@ option		:  TOK_OUTFILE '=' NAME
 			{ yyclass = xstrdup(nmstr); }
 		|  TOK_HEADER_FILE '=' NAME
 			{ headerfilename = xstrdup(nmstr); }
-	    |  TOK_TABLES_FILE '=' NAME
-            { tablesext = true; tablesfilename = xstrdup(nmstr); }
```

Comment:
This is where the `%option tables-file` value is handled, not the command-line argument.
Perhaps this is where part of your misunderstanding stems from.  The command-line argument
is handled elsewhere, but my fix does not require any changes to that part of the code.